src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java

```diff
@@ -352,14 +352,19 @@ private ResultHandle loadClass(final String className, final boolean useTccl) {
         final Class<?> primitiveType = matchPossiblyPrimitive(className);
         if (primitiveType == null) {
             if (useTccl) {
-                if (cachedTccl == null) {
-                    ResultHandle currentThread = invokeStaticMethod(THREAD_CURRENT_THREAD);
-                    cachedTccl = invokeVirtualMethod(THREAD_GET_TCCL, currentThread);
+                if (isNonPrimitiveJavaClass(className)) {
```

Comment:
Should it be `! `? Or I misunderstood your intent.

Also I think it was done at some point and @zakkak reverted it: https://github.com/quarkusio/gizmo/pull/126 .

Reply:
You are right, the check was wrong.

Reply:
And indeed https://github.com/quarkusio/gizmo/pull/126 seems to cause a problem, so let me close this

Reply:
Actually, I'll reopen this, but limit ths scope

Reply:
PR updated